src/bepasty/apis/rest.py

```diff
@@ -5,17 +5,103 @@
 
 from flask import Response, make_response, url_for, jsonify, stream_with_context, request, current_app
 from flask.views import MethodView
+from werkzeug.exceptions import HTTPException, BadRequest, Conflict, Forbidden, InternalServerError, MethodNotAllowed
 
-from ..constants import COMPLETE, FILENAME, SIZE, TYPE, TRANSACTION_ID
+from ..constants import FILENAME, ID, SIZE, TYPE, TRANSACTION_ID
 from ..utils._compat import bytes_type
 from ..utils.date_funcs import get_maxlife
 from ..utils.http import ContentRange, DownloadRange
 from ..utils.name import ItemName
-from ..utils.permissions import CREATE, LIST, READ, may
+from ..utils.permissions import CREATE, LIST, may
 from ..utils.upload import Upload, background_compute_hash
+from ..views.filelist import file_infos
+from ..views.download import DownloadView
 
 
-class ItemUploadView(MethodView):
+# This wrappper handles exceptions in the REST api implementation.
+#
+# The @blueprint.add_errorhandler decorator could do this, but we have
+# "/lodgeit/" in the same blueprint and there is no way to exclude
+# from using the same error handler ("/lodgeit/" is not REST api).
+def rest_errorhandler(func):
+    def error_message(description, code):
+        # maybe better to return error as json?
+        return description, code
+
+    def handler(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except HTTPException as e:
+            return error_message(e.description, e.code)
+        except Exception:
+            if current_app.propagate_exceptions:
+                # if testing/debug mode, re-raise
+                raise
+            exc = InternalServerError()
+            return error_message(exc.description, exc.code)
```

Comment:
outdent this line 1 level and use:
```
return exc.description, exc.code
```

Reply:
you meant the following?

```python
def rest_errorhandler(func):
    def handler(*args, **kwargs):
        try:
            return func(*args, **kwargs)
        except HTTPException as exc:
            pass
        except Exception:
            if current_app.propagate_exceptions:
                # if testing/debug mode, re-raise
                raise
            exc = InternalServerError()
        return exc.description, exc.code

    return handler
```

if so, looks like not working,
```
src/bepasty/apis/rest.py:28:9: F841 local variable 'exc' is assigned to but never used
```

Reply:
ehrm. either i do not know some strange scoping rule or this is a false positive from flake8 tester...

Reply:
it was actual error in python3 at least

Reply:
ok, looks like the first. exc is only defined inside the except block.

```
def rest_errorhandler(func):
    def handler(*args, **kwargs):
        try:
            return func(*args, **kwargs)
        except HTTPException as exc:
            return exc.description, exc.code
        except Exception:
            if current_app.propagate_exceptions:
                # if testing/debug mode, re-raise
                raise
            exc = InternalServerError()
            return exc.description, exc.code

    return handler
```